osm-fudge/__main__.py

```diff
@@ -0,0 +1,16 @@
+import bk_tree
+import sys
+
+file  = open(sys.argv[1], 'r')
```

Comment:
When you `open` a file for reading you should also `close` it to release the resources.

Because this is a common pattern to:
- initialize something, then
- work on it, then
- clean it up

Python has so called context managers to abstract over this pattern.

Maybe look up context managers and the `with` statement and re-write this to make use of it :)

Reply:
Also if you understand these open/close or in general init/cleanup patterns and context managers, now you also understand C++'s constructors and destructors :) It's exactly the same, and it allows you to couple resource management to an object's lifetime.